### PR TITLE
Adds evaluation scoring function to replicate results using BigBio / EleutherAI eval harness

### PR DESCRIPTION
**Pull request by Miking98**
Hi! I was working on a project trying to replicate numbers from this fork of the EleutherAI eval harness: [https://github.com/bigscience-workshop/lm-evaluation-harness/tree/master](https://github.com/bigscience-workshop/lm-evaluation-harness/tree/master) but for some reason wasn't able to exactly replicate with Manifest, so I wrote a couple functions to bridge those differences.

I tried to make minimal changes to the code (i.e. just adding additional endpoints), but please let me know if there's anything to change / update to bring this more in line with the rest of the repo!

### CODE REVIEW

File: manifest/api/models/huggingface.py [hunk 1]
```diff
@@ -646,3 +673,129 @@ def score_sequence(
...
+            # For gpt2-like models, we append the label to the
+            # end of the prompt, measure the entire sequence's
+            # logprob, and count that as the label's logprob
+            # (since the label is the only thing that changes
+            # between sequences, so the prompt's logprob
+            # will be constant across all labels)
+
+            # Pad right instead of left
+            encoded_prompt = self.pipeline.tokenizer(
+                [f"{x} {y}" for (x, y) in zip(prompts, labels)],
```
Comment (lorr1) [inline manifest/api/models/huggingface.py:755]: Does this assume no space on the labels? I've played with this method before and found it super sensitive to spacing of the labels? Not sure if it's a good idea to strip the labels?

File: manifest/api/models/huggingface.py [hunk 2]
```diff
@@ -646,3 +673,129 @@ def score_sequence(
...
+            label_probs = masked_probs.sum(dim=-1)
+        elif self.model_type == "text-generation":
+            # For gpt2-like models, we append the label to the
+            # end of the prompt, measure the entire sequence's
+            # logprob, and count that as the label's logprob
+            # (since the label is the only thing that changes
+            # between sequences, so the prompt's logprob
+            # will be constant across all labels)
+
+            # Pad right instead of left
```
Comment (lorr1) [inline manifest/api/models/huggingface.py:753]: Are you padding to the right here?

**Review by lorr1 — CHANGES_REQUESTED** [review]
One small change wrt to passing the PORT correctly.

But otherwise thank you so much. I'll come in afterwards and add a few tests for this.

**Comment by lorr1** [comment]
Can you also squash some of the commits with a `rebase`.

File: manifest/api/app.py [hunk 3]
```diff
@@ -100,14 +100,29 @@ def parse_args() -> argparse.Namespace:
...
+    )
     args = parser.parse_args()
     return args
 
 
 def is_port_in_use(port: int) -> bool:
     """Check if port is in use."""
     with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
-        return s.connect_ex(("localhost", port)) == 0
+        return s.connect_ex(("localhost", int(port))) == 0
```
[the remark this answers is not in the document]
Comment (Miking98) [reply manifest/api/app.py:125]: oh yep good call!

File: manifest/api/models/huggingface.py (hunk 2 above)
Comment (Miking98) [reply manifest/api/models/huggingface.py:753]: Great catch yes sorry, I had that reversed!

**Comment by Miking98** [comment]
OK just rebased to squash my commits! Please let me know what my next step is @lorr1 !

**Pull request closed without merging**